src/modules/V3ToV4Migrator.sol

```diff
@@ -4,10 +4,27 @@ pragma solidity ^0.8.0;
 import {RouterImmutables} from "../base/RouterImmutables.sol";
 import {IV3NonfungiblePositionManager} from
     "pancake-v4-periphery/src/interfaces/external/IV3NonfungiblePositionManager.sol";
+import {Actions} from "pancake-v4-periphery/src/libraries/Actions.sol";
+import {CalldataDecoder} from "pancake-v4-periphery/src/libraries/CalldataDecoder.sol";
+import {IPositionManager} from "pancake-v4-periphery/src/interfaces/IPositionManager.sol";
+import {console2} from "forge-std/console2.sol";
 
 /// @title V3 to V4 Migrator
 /// @notice A contract that migrates liquidity from PancakeSwap V3 to V4
 abstract contract V3ToV4Migrator is RouterImmutables {
+    using CalldataDecoder for bytes;
+
+    error NotAuthorizedForToken(uint256 tokenId);
+    error InvalidAction(bytes4 action);
+    error OnlyMintAllowed();
+    error OnlyAddLiqudityAllowed();
+    error BlacklistedAction();
+
+    enum PoolType {
```

Comment:
Curious is it being used somewhere ?

Reply:
my bad.. this was my POC code previously to see if it reduces the bytecode size.. will remove relevant and udpate shortly..

Reply:
updated, all the other comments should be 'resolved' now, sorry about it